openprescribing/dmd/tests.py

```diff
@@ -84,53 +89,83 @@ def test_import_dmd_snomed(self):
+    def test_fetch_and_import_ncso_concessions(self):
+        # We "download" the following concessions:
+        #  2017_11 (current)
+        #   * Amiloride (new-and-matched)
+        #   * Amlodipine (new-and-unmatched)
+        #  2017_10 (archive)
+        #   * Amiloride (changed)
+        #   * Anastrozole (unchanged)
+
         vmpp1 = DMDVmpp.objects.create(
             vppid=1092811000001107,
             nm='Amiloride 5mg tablets 100 tablet',
         )
         vmpp2 = DMDVmpp.objects.create(
-            vppid=8049011000001108,
-            nm='Duloxetine 40mg gastro-resistant capsules 56 capsule',
+            vppid=975211000001100,
+            nm='Anastrozole 1mg tablets 28 tablet',
         )
 
         NCSOConcession.objects.create(
-            year_and_month='2017_01',
-            drug='Duloxetine 40mg capsules',
-            pack_size='56',
-            price_concession_pence=600,
-            vmpp_id=vmpp2.vppid,
+            year_and_month='2017_10',
+            drug='Amiloride 5mg tablets',
+            pack_size='28',
+            price_concession_pence=925,
+            vmpp_id=vmpp1.vppid,
         )
-
-        path = 'dmd/tests/fixtures/commands/ncso_concessions_2017_02.csv'
-        call_command('import_ncso_concessions', '--filename', path)
-
-        cons1 = NCSOConcession.objects.get(
-            year_and_month='2017_02',
-            drug='Amiloride 5mg tablets'
-        )
-
-        cons2 = NCSOConcession.objects.get(
-            year_and_month='2017_02',
-            drug='Duloxetine 40mg gastro-resistant capsules'
-        )
-
-        cons3 = NCSOConcession.objects.get(
-            year_and_month='2017_02',
-            drug='Vitamin B Co Strong tablets'
+        NCSOConcession.objects.create(
+            year_and_month='2017_10',
+            drug='Anastrozole 1mg tablets',
+            pack_size='28',
+            price_concession_pence=1335,
+            vmpp_id=vmpp2.vppid,
         )
 
-        self.assertEqual(cons1.pack_size, '100')
-        self.assertEqual(cons1.price_concession_pence, 925)
-        self.assertEqual(cons1.vmpp, vmpp1)
-
-        self.assertEqual(cons2.pack_size, '56')
-        self.assertEqual(cons2.price_concession_pence, 725)
-        self.assertEqual(cons2.vmpp, vmpp2)
-
-        self.assertEqual(cons3.pack_size, '28')
-        self.assertEqual(cons3.price_concession_pence, 525)
-        self.assertEqual(cons3.vmpp, None)
+        self.assertEqual(NCSOConcession.objects.count(), 2)
+
+        base_path = os.path.join(settings.SITE_ROOT, 'dmd', 'tests', 'pages')
+
+        with open(os.path.join(base_path, 'ncso-archive.html')) as f:
+            archive_doc = bs4.BeautifulSoup(f.read(), 'html.parser')
+
+        with open(os.path.join(base_path, 'ncso-current.html')) as f:
+            current_doc = bs4.BeautifulSoup(f.read(), 'html.parser')
+
+        patch_path = 'dmd.management.commands.fetch_and_import_ncso_concessions'
+        with nested(
```

Comment:
Isn't the `nested` redundant here?

Reply:
Don't think so -- see http://www.voidspace.org.uk/python/mock/patch.html#nesting-patch-decorators.  (That said, I find the `mock` API pretty confusing so I may be missing a simpler solution.)

Reply:
I'm pretty sure that's out of date - in Python 2.7 you can do `with X as a, Y as b:`.

From [the python docs](https://docs.python.org/2/library/contextlib.html#contextlib.nested):
> This function has been deprecated in favour of the multiple manager form of the with statement.